.github/workflows/build_and_test.yml

```diff
@@ -34,23 +34,22 @@ jobs:
       - name: Install dependencies
         run: |
           apk update && apk add cppcheck python3-dev
-          python3 -m pip install cmake-format clang-format==18.1.6
+          python3 -m venv .venv
+          . .venv/bin/activate
+          python3 -m pip install setuptools clang-format==18.1.6 clang-tidy==18.1.8 pre-commit==3.8.0
```

Comment:
Maybe use the latest pre-commit 4.0.1?

Reply:
I removed version. I think it is better because pre-commit itself can be updated since this won't affect the formatting.

Reply:
I think it's always better to stick with the specific version of the CI. We don't know whether the future version can still be compatible with the current files.

Reply:
If pre-commit doesn't have much breaking changes, then it is fine to stick with the latest version.